app/Actions/Site/UpdateEnv.php

```
@@ -39,59 +39,125 @@ public function update(Site $site, array $input): void
      * @return array<int, array{key: string, value: string, is_secret: bool}>
+     *
+     * @throws SSHError
      */
-    private function processVariables(Site $site, array $input): array
+    private function resolveVariables(Site $site, array $input, string $path): array
     {
+        $secretKeys = array_flip(EnvParser::secretKeys($site->env_variables));
+
         if (isset($input['variables']) && is_array($input['variables'])) {
-            $incoming = array_map(function ($var) {
+            $live = EnvParser::parse($site->getEnv($path));
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Helpers\EnvParser' in method 'resolveVariables'. (undefined)

(StaticAccess)